graphql/schema/types/user.graphql

```diff
@@ -0,0 +1,76 @@
+enum RoleEnum {
+    READ
+    VOTE
+    EDIT
+    MODIFY
+    ADMIN
+}
+
+type User {
+    id: ID!
+    name: String!
+    roles: [RoleEnum!]!
+    # TODO - should this be mandatory?
+    email: String
```

Comment:
I think it's infeasible to run a website with accounts without email registration. It would make password resets impossible, and would make it too easy to create spam/sockpuppet accounts. protonmail is free, I don't think it's that big of a burden.

Reply:
You make a decent point. If I were administering such a system, I'd like it invite-based to sidestep the whole spam/sockpuppet issue, but I realise that such a system may not scale well. I'd left it open because I wasn't sure if people were averse to providing an email address because of the subject matter. In any case, I think you make a good case and I'll change it to be mandatory.